active_linters_view.py

```diff
@@ -25,33 +28,57 @@
 State = {
     'assigned_linters_per_file': defaultdict(set),
     'failed_linters_per_file': defaultdict(set),
-    'problems_per_file': defaultdict(dict)
+    'problems_per_file': defaultdict(dict),
+    'sweeps': defaultdict(int),
+    'expanded_ok': defaultdict(float),
 }  # type: State_
 
 
 def plugin_unloaded():
     events.off(redraw_file)
+    events.off(on_finished_linting)
 
     for window in sublime.windows():
         for view in window.views():
             view.erase_status(STATUS_ACTIVE_KEY)
 
 
+@events.on(events.LINT_END)
+def on_finished_linting(buffer_id, **kwargs):
+    # run on the main thread to avoid locks; all other
+    # functions mutating "sweeps" or "expanded_ok" already
+    # run on the ui thread
+    def task():
+        State['sweeps'][buffer_id] += 1
+        if State['sweeps'][buffer_id] == 1:
+            show_expanded_ok(buffer_id)
+    sublime.set_timeout(task)
+
+
 @events.on(events.LINT_RESULT)
 def redraw_file(filename, linter_name, errors, **kwargs):
+    # type: (Filename, LinterName, List[persist.LintError], object) -> None
     problems = State['problems_per_file'][filename]
     if linter_name in State['failed_linters_per_file'][filename]:
-        problems[linter_name] = '(erred)'
+        problems[linter_name] = '?'
     elif linter_name in State['assigned_linters_per_file'][filename] or errors:
         if linter_name not in State['assigned_linters_per_file'][filename]:
             State['assigned_linters_per_file'][filename].add(linter_name)
 
-        we_count = count_problems(errors)
-        if we_count == (0, 0):
-            problems[linter_name] = '(ok)'
+        counts = count_problems(errors)
+        if sum(counts.values()) == 0:
+            problems[linter_name] = ''
         else:
-            tpl = persist.settings.get('statusbar.counters_template')
-            problems[linter_name] = tpl.format(*we_count)
+            sorted_keys = (
+                tuple(sorted(counts.keys() - {'w', 'e'}))
+                + ('w', 'e')
+            )
+            parts = ' '.join(
+                "{}:{}".format(error_type, counts[error_type])
+                for error_type in sorted_keys
+                if error_type in counts and counts[error_type] > 0
+            )
+            problems[linter_name] = '({})'.format(parts)
```

Comment:
I think we might as well do without the parentheses here. Go big or go home. It's been a while since we did a big release that changed stuff. Changing this will also signal to users that we made a serious redesign of the status bar info and not just took away their template setting.

Reply:
![image](https://user-images.githubusercontent.com/8558/80144235-727fc000-85ae-11ea-94be-bc8233c307c7.png)

![image](https://user-images.githubusercontent.com/8558/80144252-77dd0a80-85ae-11ea-9942-2ab3b7038261.png)

Well, I rather go home before breaking.  

Just removing the parentheses actually feels like going back to step 0, the design phase.  Since I removed all other delimiters (no ", " between the linters), we start all over again.  "mypy n:2 e:3 annotation w:3"? It is really easy right now to scan for the "("/")" signs.  Especially round parentheses (circles, badges) are very common design lang for this, "Inbox (1)" usually is used for one unread message, whereby "Inbox 1" indicates its name, like there is also an "Inbox 2".

Typical badge, of course with lots of whitespace because the web always uses lots of whitespace, basically the only currently known design language is whitespace: 
![image](https://user-images.githubusercontent.com/8558/80890889-a353a980-8cc0-11ea-88bc-856eeae87380.png)

Reply:
Oh, that's a really good example and a very good reason for keeping the parentheses. I totally agree now 👍